src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -109,6 +117,10 @@ private static Type LookForTypeInAssemblies(TypeName typeName, IEnumerable<Assem
                             }
                         }
                     }
+                    else
+                    {
+                        searchedAssemblies.Add(assembly.FullName);
```

Comment:
Isn't line 89 saying - type wasn't found and we have another name to look for? And if we find that alternate name, you aren't adding anything to `searchedAssemblies`.